### PR TITLE
Fix: Avoid error from null value for observer in timeoutDisconnect function

### PR DESCRIPTION
**Pull request by KvNGCzA**
*Issue #, if available:*
#515 #544 #596 

*Description of changes:*
An if statement to make sure that the `observer` exists before it tries to call the properties of `observer` in timeoutDisconnect function.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by gprasadholla** [comment]
Anything holding this PR back? I have been seeing this error lot of times and getting picked by crashlytics. we should merge this sooon

**Comment by KvNGCzA** [comment]
@elorzafe please check this out for review

**Comment by KvNGCzA** [comment]
@elorzafe Thank you. This PR exists for a similar issue. Please spare a few minutes to check it out

File: packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts [hunk 1]
```diff
@@ -696,6 +696,10 @@ export class AppSyncRealTimeSubscriptionHandshakeLink extends ApolloLink {
 
   private _timeoutDisconnect() {
     this.subscriptionObserverMap.forEach(({ observer }) => {
+      if (!observer) {
+        return;
+      }
+
```
Comment (elorzafe) [inline packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts:702]: @KvNGCzA I think this should never happen. If the map is empty, the callback wouldnt been call? Did you experience something that the observer is `undefined`. I think the issue is somewhere else. I will keep looking

**Comment by elorzafe** [comment]
@KvNGCzA @gprasadholla we just published the first fix, next week I will continue working on the other issues and doing a following release.

File: packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts (hunk 1 above)
Comment (randomslap) [reply packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts:702]: @elorzafe After a subscription is established, I randomly get the following error(s): `Undefined is not an object evaluating (_a.observer)` or `Cannot read property 'observer' of undefined`

These errors appear randomly in my case. I'm suspecting it has to do with something with the subscription timing out.

Comment (gprasadholla) [reply packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts:702]: I was able to reproduce this error by randomly switching between networks(WiFi/4G). I have already rollout a build with this patch and I am no longer getting this crash. Thank you

**Review by elorzafe — APPROVED** [review]
Thanks @KvNGCzA ! 🥇

**Comment by manueliglesias** [comment]
This was addressed by #616

**Pull request closed without merging**